src/api/heap_profiler.rs

```diff
@@ -23,20 +25,43 @@ impl HeapProfilerState {
             memory: HashMap::new(),
             started: SystemTime::now(),
             heap_history: VecDeque::new(),
+            processes: vec![],
         }
     }
 
-    // Merge child process profile results into parent process profile.
+    pub fn add_process(&mut self, process: <Self as HostFunctions>::Wrap) {
+        self.processes.push(process);
+    }
+
+    // Collect all process profiles into parent profile and remove them from process list
     //
-    // Usually this should be called when child process exits.
+    // Safety: this is safe to call only when all processes are finished.
+    pub fn collect_data(&mut self) -> anyhow::Result<()> {
```

Comment:
This is only one strategy, used for merging and writing merged profile to fs. We could implement other ways to use and show heap profile later